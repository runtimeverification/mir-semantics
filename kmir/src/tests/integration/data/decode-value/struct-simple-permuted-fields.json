{
  "bytes": [
    1,
    15,
    31
  ],
  "types": [
    [
      0,
      {
        "PrimitiveType": {
          "Uint": "U8"
        }
      }
    ],
    [
      1,
      {
        "PrimitiveType": "Bool"
      }
    ]
  ],
  "typeInfo": {
    "StructType": {
      "name": "core::ops::RangeInclusive<u8>",
      "adt_def": 100,
      "fields": [
        0,
        0,
        1
      ],
      "layout": {
        "fields": {
          "Arbitrary": {
            "offsets": [
              {
                "num_bits": 8
              },
              {
                "num_bits": 16
              },
              {
                "num_bits": 0
              }
            ]
          }
        },
        "variants": {
          "Single": {
            "index": 0
          }
        },
        "abi": {
          "Aggregate": {
            "sized": true
          }
        },
        "abi_align": 1,
        "size": {
          "num_bits": 24
        }
      }
    }
  }
}
