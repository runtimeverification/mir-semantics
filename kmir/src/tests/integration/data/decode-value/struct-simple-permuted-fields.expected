Aggregate ( variantIdx ( 0 ) , ListItem ( Integer ( 15 , 8 , false ) )
ListItem ( Integer ( 31 , 8 , false ) )
ListItem ( BoolVal ( true ) ) )
